activestorage/app/models/active_storage/attachment.rb

```diff
@@ -74,7 +74,7 @@ def purge_dependent_blob_later
     end
 
     def dependent
-      record.attachment_reflections[name]&.options[:dependent]
+      record.attachment_reflections[name]&.options&.[](:dependent)
```

Comment:
Personally I find `.fetch` or `.dig` more readable for such cases
```suggestion
      record.attachment_reflections[name]&.options&.fetch(:dependent, nil)
```

Reply:
Thats fair, using the safe navigator with operations looks really tacky.